components/automate-ui/src/app/pages/user-details/user-details.component.ts

```diff
@@ -75,7 +75,11 @@ export class UserDetailsComponent implements OnInit, OnDestroy {
     // TODO (tc) This needs to be refactored to resemble our other patterns
     // for specific object pages.
     // combineLatest depends on the user object existing already.
-    this.user = <User>{};
+    this.user = {
```

Comment:
Worth changing the user interface to something like
```typescript
interface User {
  id?: string;
  name?: string;
  membership_id?: string;
}
```
...so we could not set them to empty strings here?

Reply:
Here is my reasoning: in any strongly-typed language, if you have a `User` object that contains a property `name`, you know with certainty that the property `user.name` exists. But with the original code (`user = <User>{};`), or with your suggested change, we no longer have that certainty. That may or may not have any practical impact; referencing the non-existent `user.name` would return undefined instead of an empty string. For a non-string field, it becomes more problematic. Say we had a type field that could be 'a' or 'b' only, but making it optional wouldn't undefined also need to be checked? That seems akin to the Worst Mistake in Computer Science (e.g. https://bit.ly/1JHIqbY), no? TypeScript lets us be tolerant, though, as you have suggested... but should we be? Not sure. 🤷‍♂

Reply:
> in any strongly-typed language, if you have a User object that contains a property name, you know with certainty that the property user.name exists.

...but if we set `user.name` to `''`, what's certainty isn't worth much: we don't know if it's a valid, proper username -- we only know it's not `undefined`. That's not the kind of certainty I appreciate :smile: 

That said, there's surely better ways than what I have proposed. Akin to a `Maybe` or `Optional` type, we might be able to say `User | undefined`. I've never done a deep dive into how to do this in TS _The Right Way_; I'm vaguely aware that there's [a project called `ts-fp` that provides `Option`](https://gcanti.github.io/fp-ts/modules/Option.ts.html).

Reply:
Why does `this.user` need to be initialized here?

Reply:
@bcmdarroch: I was trying to answer that very question above; sorry if not clear. In a nutshell:
1. If you have something that is a User object, you might assign null to it, in which case it certainly does not have an id, name, etc.
2. But if it is non-null, I do not believe one should have to wonder (read: check or worse, blow-up because you did not check) if it has a property named `id`. It is a User object: it should _always_ have that property. Now, whether that property itself is initialized--to @srenatus 's point--is worth considering, too, but I'm just taking the first, small step here.

Reply:
> but I'm just taking the first, small step here.

Let's not spread this, but keep trying to improve it in each step, then 😃